agent/handlers/task_server_setup_test.go

```diff
@@ -3561,3 +3566,114 @@ func TestUpdateTaskProtection(t *testing.T) {
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.BlackHolePortFaultType,
+			method:             "PUT",
+		},
+		{
+			name:               "DELETE blackholeport",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.BlackHolePortFaultType,
+			method:             "DELETE",
+		},
+		{
+			name:               "GET blackholeport",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.BlackHolePortFaultType,
+			method:             "GET",
+		},
+		{
+			name:               "PUT latency",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.LatencyFaultType,
+			method:             "PUT",
+		},
+		{
+			name:               "DELETE latency",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.LatencyFaultType,
+			method:             "DELETE",
+		},
+		{
+			name:               "GET latency",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.BlackHolePortFaultType,
+			method:             "GET",
+		},
+		{
+			name:               "PUT packet loss",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.PacketLossFaultType,
+			method:             "PUT",
+		},
+		{
+			name:               "DELETE packet loss",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.PacketLossFaultType,
+			method:             "DELETE",
+		},
+		{
+			name:               "GET packet loss",
+			expectedStatusCode: http.StatusOK,
+			fault:              faulttype.PacketLossFaultType,
+			method:             "GET",
+		},
+		{
+			name:               "PUT unknown",
+			expectedStatusCode: http.StatusNotFound,
+			fault:              "unknown",
+			method:             "PUT",
+		},
+		{
+			name:               "DELETE unknown",
+			expectedStatusCode: http.StatusNotFound,
+			fault:              "unknown",
+			method:             "DELETE",
+		},
+		{
+			name:               "GET unknown",
+			expectedStatusCode: http.StatusNotFound,
+			fault:              "unknown",
+			method:             "GET",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// Mocks
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			state := mock_dockerstate.NewMockTaskEngineState(ctrl)
+			statsEngine := mock_stats.NewMockEngine(ctrl)
+			ecsClient := mock_ecs.NewMockECSClient(ctrl)
+
+			agentState := agentV4.NewTMDSAgentState(state, statsEngine, ecsClient, clusterName, availabilityzone, vpcID, containerInstanceArn)
+			metricsFactory := mock_metrics.NewMockEntryFactory(ctrl)
+
+			router := mux.NewRouter()
+
+			registerFaultHandlers(router, agentState, metricsFactory)
+			var requestBody io.Reader
+			req, err := http.NewRequest(tc.method, fmt.Sprintf("/api/%s/fault/v1/%s", endpointId, tc.fault),
+				requestBody)
+			require.NoError(t, err)
+
+			// Send the request and record the response
+			recorder := httptest.NewRecorder()
+			router.ServeHTTP(recorder, req)
```

Comment:
How will we update this test when the handlers are implemented?

Reply:
Good question, I was thinking of introducing different tests functions for each of the request types (i.e. one for Start blackhole port, Stop blackhole port, and check blackhole port) in future PRs. This test was more so of actually testing whether or not the handlers can be registered to the router. I can create separate test cases in this PR if that makes more sense.